subsys/bluetooth/controller/power_clock_driver.c

```diff
@@ -10,20 +10,53 @@
 #include <kernel_includes.h>
 #include <clock_control.h>
 #include <ble_controller_soc.h>
+#include "multithreading_lock.h"
+
+static inline int ble_controller_hf_clock_request_wlock(
+	ble_controller_hf_clock_callback_t on_started)
+{
+	s32_t errcode;
+	API_LOCK_AND_RETURN_ON_FAIL;
+	errcode = ble_controller_hf_clock_request(on_started);
+	API_UNLOCK;
+
+	return errcode;
+}
+
+static inline int ble_controller_hf_clock_is_running_wlock(bool *p_is_running)
+{
+	s32_t errcode;
+	API_LOCK_AND_RETURN_ON_FAIL;
+	errcode = ble_controller_hf_clock_is_running(p_is_running);
+	API_UNLOCK;
+
+	return errcode;
+}
+
+static inline int ble_controller_hf_clock_release_wlock(void)
+{
+	s32_t errcode;
+	API_LOCK_AND_RETURN_ON_FAIL;
+	errcode = ble_controller_hf_clock_release();
+	API_UNLOCK;
+
+	return errcode;
+}
 
 static int hf_clock_start(struct device *dev, clock_control_subsys_t sub_system)
 {
 	ARG_UNUSED(dev);
 
-	if (ble_controller_hf_clock_request(NULL) != 0) {
+	if (ble_controller_hf_clock_request_wlock(NULL) != 0) {
```

Comment:
this `_wlock()` pattern appears several places where more-or-less the same function is defined again. Consider generalizing this to remove all the copy-pasted code.

Reply:
Removed these and updated the code as appropriate.